rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java

```diff
@@ -154,6 +155,54 @@ public boolean isDependencyTag(String groupId, String artifactId) {
         return false;
     }
 
+    /**
+     * Is a tag a dependency that matches the group and artifact?
+     *
+     * @param groupId    The group ID glob expression to compare the tag against.
+     * @param artifactId The artifact ID glob expression to compare the tag against.
+     * @return true if the tag matches.
+     */
+    public boolean isRequestedDependencyTag(String groupId, String artifactId) {
```

Comment:
This method and the one before also share 95% or the same code base. I would create another function:

```java
private boolean isDependencyTag(String groupId, String artifactId, Map<Scope, List<GroupArtifactVersion>> dependencies) { 
  // exact same implementation without the `Map<Scope, List<Dependency>> dependencies` line
}
```

The only thing you have to do then is map the ResolvedDependencies en Dependencies to the GroupArtifactVersion and call that method.

Reply:
I think we should keep these 2 concepts separate explicitly

Reply:
Why? I think they are quite similar, but just use a different set. What's the reason to duplicate code?